### ISSUE TITLE
Failing erfc() test case

### ISSUE THREAD

**Issue by ckormanyos** [BoostGSoC21/multiprecision#144]
Hi Fahad (@sinandredemption)

Here is another unique failing test case.

It is a test failure on the `erfc()` function.

In this test case, the special function implementation computes an extreme `exp()` function which is very near the lower range of the data type. The kind of failure is precision loss in the numerical result.

I am unsure if the underlying `cpp_double_fp< 8-bate-double >` is capable of computing this exponent to full precision for the type.

Cc: @cosurgi

```
#include <iostream>

#include <boost/math/special_functions/erf.hpp>
#include <boost/multiprecision/cpp_double_fp.hpp>
#include <boost/multiprecision/cpp_dec_float.hpp>

int main()
{
  using boost::multiprecision::cpp_double_double;

  using float_type = typename cpp_double_double::backend_type::float_type;

  using dec_float_type = boost::multiprecision::number<boost::multiprecision::cpp_dec_float<101>, boost::multiprecision::et_off>;

  // N[Erfc[25813503265380859375 / 1000000000000000000], 101]
  // 8.9699910066184047911669732896043282222388381719332458957375089781363329036291716592940465828282353321*10^-292

  const auto erfc1 = boost::math::erfc(cpp_double_double("25.813503265380859375"));
  const auto erfc2 = boost::math::erfc(dec_float_type   ("25.813503265380859375"));

  const auto ctrl1 = cpp_double_double("0.89699910066184047911669732896043282222388381719332458957375089781363329036291716592940465828282353321e-291");
  const auto ctrl2 = dec_float_type   ("0.89699910066184047911669732896043282222388381719332458957375089781363329036291716592940465828282353321e-291");

  const auto delta1 = cpp_double_double(fabs(1 - fabs(erfc1 / ctrl1)));
  const auto delta2 = dec_float_type   (fabs(1 - fabs(erfc2 / ctrl2)));

  const auto flg = std::cout.flags();

  std::cout << std::setprecision(static_cast<std::streamsize>(std::numeric_limits<cpp_double_double>::digits10)) << erfc1 << std::endl;
  std::cout << std::setprecision(static_cast<std::streamsize>(std::numeric_limits<dec_float_type   >::digits10)) << erfc2 << std::endl;

  std::cout << std::setprecision(static_cast<std::streamsize>(std::numeric_limits<cpp_double_double>::digits10)) << ctrl1 << std::endl;
  std::cout << std::setprecision(static_cast<std::streamsize>(std::numeric_limits<dec_float_type   >::digits10)) << ctrl2 << std::endl;

  std::cout << std::setprecision(static_cast<std::streamsize>(std::numeric_limits<cpp_double_double>::digits10)) << delta1 << std::endl;
  std::cout << std::setprecision(static_cast<std::streamsize>(std::numeric_limits<dec_float_type   >::digits10)) << delta2 << std::endl;

  std::cout.flags(flg);
}

```

**Comment by ckormanyos** [comment]
> computes an extreme `exp()` function

The value being computed is

$${\phantom{.}}e^{-666.337{\ldots}}{\phantom{.}}{\approx}{\phantom{.}}4.107{\times}10^{-290}$$

which is close to the upper range of this function. The exponent is computed with a positive argument first and then inverted. I'm not sure where the procision loss occurs, either in the functional exponentiation, or the inversoin. It could also be within the Pade approximation that is used for this function, since we have not done extended testing of the `exp()` function as of yet.

See also [this link](https://www.wolframalpha.com/input?i=N%5BExp%5B-666337%2F1000%5D%2C+31%5D&key=g4h5d).

```
N[Exp[-666337/1000], 31]
```

Cc: @cosurgi and @sinandredemption

**Comment by ckormanyos** [comment]
See also 

```
N[Erfc[25813503265380859375 / 1000000000000000000], 101]
```

which can be computed conveniently at Wolfram Alpha(R) [here](https://www.wolframalpha.com/input?i=N%5BErfc%5B25813503265380859375+%2F+1000000000000000000%5D%2C+101%5D&key=euhi6).

**Comment by ckormanyos** [comment]
I just investigated further.

- Using standard `eval_exp()` from the default-synthesized `number` implementation leads to similar precision difference.
- Calculating `exp(cpp_double_double(666337) / 1000)` produced accurate result.

It appears to be the string-read of the control value that is losing precision.

**Comment by ckormanyos** [comment]
Hi Fahad (@sinandredemption) the root cause can be traced to #145 

Cc: @cosurgi

**Comment by ckormanyos** [comment]
Fixed by #146
